text/0000-build-std.md

```diff
@@ -0,0 +1,2390 @@
+    workspace and build-std does not need to generate one
+    - On-disk lockfile is now used so do not need to worry about it changing
+  - Still need to implement `--locked` behaviour
+    - Doing so would break `[patch]` with build-std
+- **[wg-cargo-std-aware#39]: Figure out interaction with public/private dependencies**, [ehuss], Sep 2019
+  - Public/private status of the standard library is hardcoded to public in MVP - should
+    it be?
+  - If defined in `Cargo.toml`, it will likely inherit visibility from that, which defaults
+    to private.
+- **[wg-cargo-std-aware#43]: What will the UX be for enabling "build libstd"
+  logic?**, [alexcrichton], Sep 2019
+  - How will build-std eventually be enabled?
+  - At time of writing, `-Zbuild-std=$crates`, but what should the eventual
+    syntax be?
+    - e.g. `build.std` option in `Cargo.toml` or `.config/cargo`
+  - Opting into build-std as an unstable feature should be a `-Z` flag or entry
+    in `cargo-features`
+  - Eventually not a flag at all, and Cargo should try to compile the standard
+    library automatically if a compiled standard library is not available
+    - Concern that the pre-compiled standard library may not match by default
+      and there would be too many false-positive rebuilds
+    - For missing targets, need to distinguish between forgetting to run
+      `rustup target add` and needing to compile the standard library
+    - `[profile]` could not affect the standard library by default
+      - There are cases where implicit rebuilding could be desirable
+        - e.g. ABI-modifying flags
+  - Passing a flag on every invocation is not a good user experience
+  - [cargo#10308] is a prototype with explicit standard library dependencies
+    - Idea posted on internals, see *Build-std and the standard library* below
+  - If enabled manually, [cargo#8733] expressed the need to enable build-std
+    only for certain targets, in particular when cross-compiling to no-std
+    targets
+- **[wg-cargo-std-aware#46]: How to handle special object files?**, [ehuss], Sep 2019
+  - How to handle pre-built object files needed to link on a target?
+    - e.g. `crt1.o`, `crti.o`, `crtn.o`, etc for musl or `dllcrt2.o`,
+      `crt2.o`, etc for Windows GNU
+  - Cargo is target agnostic and does not want to hardcode logic for particular
+    targets
+    - The target definitions define which files they expect to find and link
+      against
+  - [rust#68887] adds a `self-contained` directory which could help
+- **[wg-cargo-std-aware#47]: how to handle pre-built linkers?**, [ehuss], Sep
+  2019
+  - Some pre-built targets ship with a copy of rust-lld or gcc to assist with
+    linking
+  - `rust-lld` could be shipped as a rustup component or the user could be
+    forced to install these components
+    - Since this issue was filed, `rust-lld` is now always shipped
+  - rustc finds `rust-lld` via `--sysroot` and if the sysroot is not provided
```

Comment:
Is this true? I think Cargo doesn't normally pass `--sysroot` and rustc finds it itself 

https://github.com/rust-lang/rust/blob/0d6ab209c525e276cbe7544cbd39a3c3619b6b18/compiler/rustc_session/src/config.rs#L2693